packages/mui-base/src/Select/Positioner/useSelectPositioner.tsx

```diff
@@ -0,0 +1,254 @@
+import * as React from 'react';
+import type {
+  VirtualElement,
+  Side,
+  Padding,
+  Boundary,
+  FloatingRootContext,
+  FloatingContext,
+  Middleware,
+} from '@floating-ui/react';
+import type { GenericHTMLProps } from '../../utils/types';
+import { useAnchorPositioning } from '../../utils/useAnchorPositioning';
+import { mergeReactProps } from '../../utils/mergeReactProps';
+import { useSelectRootContext } from '../Root/SelectRootContext';
+import { useScrollLock } from '../../utils/useScrollLock';
+
+/**
+ *
+ * API:
+ *
+ * - [useSelectPositioner API](https://mui.com/base-ui/api/use-select-positioner/)
+ */
+export function useSelectPositioner(
+  params: useSelectPositioner.Parameters,
+): useSelectPositioner.ReturnValue {
+  const { open = false, keepMounted } = params;
+
+  const { touchModality, alignMethod, innerFallback, mounted } = useSelectRootContext();
+
+  const itemAligned = alignMethod === 'item' && !innerFallback && !touchModality;
+
+  useScrollLock(itemAligned && mounted);
+
+  const {
+    positionerStyles,
+    arrowStyles,
+    hidden,
+    arrowRef,
+    arrowUncentered,
+    renderedSide,
+    renderedAlignment,
+    positionerContext: floatingContext,
+    isPositioned,
+  } = useAnchorPositioning({
+    ...params,
+    positionStrategy: itemAligned ? 'fixed' : params.positionStrategy,
+    innerOptions: {
+      fallback: params.innerFallback,
+      touchModality,
+    },
+    trackAnchor: !itemAligned,
+    collisionPadding:
+      touchModality && params.collisionPadding == null ? 20 : params.collisionPadding,
+  });
+
+  const getPositionerProps: useSelectPositioner.ReturnValue['getPositionerProps'] =
+    React.useCallback(
+      (externalProps = {}) => {
+        const hiddenStyles: React.CSSProperties = {};
+
+        if ((keepMounted && !open) || hidden) {
+          hiddenStyles.pointerEvents = 'none';
+        }
+
+        return mergeReactProps(externalProps, {
+          tabIndex: -1,
+          inert: open ? undefined : 'true',
+          style: {
+            ...positionerStyles,
+            ...hiddenStyles,
+            zIndex: 2147483647, // max z-index
```

Comment:
This value is used several times across the codebase. Should be extracted to a const.